WordPress/gutenberg
# Support [nested] lists in table cells by reformatting simple content to appear more list-like.

**Note**: This PR needs some more test cases, and some edge cases could be improved, but I'd like to get feedback on the general approach. 

Related to https://github.com/WordPress/gutenberg/issues/45774



## What?
It's based on requirement 2 from https://github.com/WordPress/gutenberg/issues/45774. It allows `ul` and `ol` elements to be pasted into tables. These elements are converted to simple content and whitespace is used to make them look like the source lists. 

## Why?

Increase parity between Gutenberg and other editors. 

## How?

Allow the list schema to be embedded into `td` and `th`. Add a `transform` function for `tables` that can reformat lists. 

## Testing Instructions




1. Check out this branch and create a table in Google Docs that has ordered or unordered in its cells. See the screenshots below. 
2. Paste the table into Gutenberg to see the result
3. Try adding content before/after the list.
4. Try nesting lists to any depth. 


### Testing Instructions for Keyboard


## Screenshots or screencast 


Google Docs table with lists

![image](https://user-images.githubusercontent.com/6851384/205153285-e015f851-f67c-4905-aedd-aea0a0464383.png)

Gutenberg

![image](https://user-images.githubusercontent.com/6851384/205153492-f0f6335f-9707-497a-8427-227037868cbe.png)

Note that the two arrows. This content was added to a `li` by using `shift` + `enter`. Ideally, they should be indented so they look closer to the source content.
